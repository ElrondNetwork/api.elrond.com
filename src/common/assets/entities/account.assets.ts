import { Field, ObjectType } from "@nestjs/graphql";

@ObjectType("AccountAssets", { description: "Account assets object type." })
export class AccountAssets {
  @Field(() => String, { description: "Name for the given account asset." })
  name: string = '';

  @Field(() => String, { description: "Description for the given account asset." })
  description: string = '';

  @Field(() => [String], { description: "Tags list for the given account asset." })
  tags: string[] = [];

  @Field(() => String, { description: "Proof for the given account asset." })
  proof: string | undefined = undefined;

  @Field(() => String, { description: "Icon for the given account asset." })
  icon: string | undefined = undefined;

  @Field(() => String, { description: "Icon PNG link for the given account asset." })
  iconPng: string | undefined = undefined;

  @Field(() => String, { description: "Icon SVG link for the given account asset." })
  iconSvg: string | undefined = undefined;
}
