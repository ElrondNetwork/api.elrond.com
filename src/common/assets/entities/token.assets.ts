import { ApiProperty } from "@nestjs/swagger";
import { TokenAssetStatus } from "../../../endpoints/tokens/entities/token.asset.status";

export class TokenAssets {
  @ApiProperty({ type: String })
  website: string = '';

  @ApiProperty({ type: String })
  description: string = '';

  @ApiProperty({ enum: TokenAssetStatus, default: 'inactive' })
  status: TokenAssetStatus = TokenAssetStatus.inactive;

  @ApiProperty({ type: String })
  pngUrl: string = '';

  @ApiProperty({ type: String })
  svgUrl: string = '';

  @ApiProperty({ type: String, isArray: true })
  lockedAccounts: string[] | Record<string, string> | undefined = undefined;

  @ApiProperty({ type: String, isArray: true })
  extraTokens: string[] | undefined = undefined;
}
