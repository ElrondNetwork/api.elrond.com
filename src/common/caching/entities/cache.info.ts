import { Constants } from "src/utils/constants";

export class CacheInfo {
  key: string = "";
  ttl: number = Constants.oneSecond() * 6;

  static Nodes: CacheInfo = {
    key: 'nodes',
    ttl: Constants.oneHour(),
  };

  static NumShards: CacheInfo = {
    key: 'numShards',
    ttl: Constants.oneWeek(),
  };


  static GenesisTimestamp: CacheInfo = {
    key: 'genesisTimestamp',
    ttl: Constants.oneWeek(),
  };

  static ActiveShards: CacheInfo = {
    key: 'shards',
    ttl: Constants.oneMinute(),
  };

  static AllEsdtTokens: CacheInfo = {
    key: 'allEsdtTokens',
    ttl: Constants.oneHour(),
  };

  static Identities: CacheInfo = {
    key: 'identities',
    ttl: Constants.oneMinute() * 15,
  };

  static Providers: CacheInfo = {
    key: 'providers',
    ttl: Constants.oneHour(),
  };

  static ProvidersWithStakeInformation: CacheInfo = {
    key: 'providersWithStakeInformation',
    ttl: Constants.oneHour(),
  };

  static Keybases: CacheInfo = {
    key: 'keybases',
    ttl: Constants.oneHour(),
  };

  static KeybaseConfirmation(keybase: string): CacheInfo {
    return {
      key: `keybase:${keybase}`,
      ttl: Constants.oneMonth() * 6,
    };
  }

  static TxCount(address: string): CacheInfo {
    return {
      key: `txCount:${address}`,
      ttl: Constants.oneSecond() * 30,
    };
  }

  static IdentityProfilesKeybases: CacheInfo = {
    key: 'identityProfilesKeybases',
    ttl: Constants.oneHour(),
  };

  static IdentityProfile(key: string): CacheInfo {
    return {
      key: `identityProfile:${key}`,
      ttl: Constants.oneMonth() * 6,
    };
  }

  static CurrentPrice: CacheInfo = {
    key: 'currentPrice',
    ttl: Constants.oneHour(),
  };

  static Economics: CacheInfo = {
    key: 'economics',
    ttl: Constants.oneMinute() * 10,
  };

  static Top25Accounts: CacheInfo = {
    key: 'accounts:0:25',
    ttl: Constants.oneMinute() * 2,
  };

  static ShardAndEpochBlses(shard: any, epoch: any): CacheInfo {
    return {
      key: `${shard}_${epoch}`,
      ttl: Constants.oneWeek(),
    };
  }

  static OwnerByEpochAndBls(bls: string, epoch: number): CacheInfo {
    return {
      key: `owner:${epoch}:${bls}`,
      ttl: Constants.oneDay(),
    };
  }

  static ShardNonce(shard: number): CacheInfo {
    return {
      key: `shardNonce:${shard}`,
      ttl: Number.MAX_SAFE_INTEGER,
    };
  }

  static TokenAssets: CacheInfo = {
    key: 'tokenAssets',
    ttl: Constants.oneDay(),
  };

  static TokenTransferProperties(identifier: string): CacheInfo {
    return {
      key: `token:transfer:properties:v2:${identifier}`,
      ttl: Constants.oneDay(),
    };
  }

  static EsdtProperties(identifier: string): CacheInfo {
    return {
      key: `esdt:${identifier}`,
      ttl: Constants.oneDay(),
    };
  }

  static EsdtAddressesRoles(identifier: string): CacheInfo {
    return {
      key: `esdt:roles:${identifier}`,
      ttl: Constants.oneDay(),
    };
  }

  static EsdtAssets(identifier: string): CacheInfo {
    return {
      key: `esdt:assets:${identifier}`,
      ttl: Constants.oneDay(),
    };
  }

  static NftMetadata(identifier: string): CacheInfo {
    return {
      key: `nftMetadata:${identifier}`,
      ttl: Constants.oneHour(),
    };
  }

  static NftMedia(identifier: string): CacheInfo {
    return {
      key: `nftMedia:${identifier}`,
      ttl: Constants.oneHour(),
    };
  }

  static NftMediaProperties(uri: string): CacheInfo {
    return {
      key: `nftMediaProperties:${uri}`,
      ttl: Constants.oneHour(),
    };
  }

  static TokenTransactions(identifier: string): CacheInfo {
    return {
      key: `tokenTransactions:${identifier}`,
      ttl: Constants.oneMinute(),
    };
  }

  static TokenHolders(identifier: string): CacheInfo {
    return {
      key: `tokenHolders:${identifier}`,
      ttl: Constants.oneMinute(),
    };
  }
}