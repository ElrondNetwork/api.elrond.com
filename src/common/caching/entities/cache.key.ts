import { Constants } from "src/utils/constants";

export class CacheKey {
  key?: string;
  ttl: number = Constants.oneSecond() * 6;

  static Nodes(): CacheKey {
    return {
      key: 'nodes',
      ttl: Constants.oneHour()
    }
  }

  static AllEsdtTokens(): CacheKey {
    return {
      key: 'allEsdtTokens',
      ttl: Constants.oneHour()
    }
  }

  static Identities(): CacheKey {
    return {
      key: 'identities',
      ttl: Constants.oneMinute() * 15
    }
  }

  static Providers(): CacheKey {
    return {
      key: 'providers',
      ttl: Constants.oneHour()
    }
  }

  static ProvidersWithStakeInformation(): CacheKey {
    return {
      key: 'providersWithStakeInformation',
      ttl: Constants.oneHour()
    }
  }

  static Keybases(): CacheKey {
    return {
      key: 'keybases',
      ttl: Constants.oneHour()
    }
  }

  static IdentityProfilesKeybases(): CacheKey {
    return {
      key: 'identityProfilesKeybases',
      ttl: Constants.oneHour()
    }
  }

  static CurrentPrice(): CacheKey {
    return {
      key: 'currentPrice',
      ttl: Constants.oneHour()
    }
  }

  static Economics(): CacheKey {
    return {
      key: 'economics',
      ttl:  Constants.oneMinute() * 10
    }
  }

  static Top25Accounts(): CacheKey {
    return {
      key: 'accounts:0:25',
      ttl:  Constants.oneMinute() * 2
    }
  }
}