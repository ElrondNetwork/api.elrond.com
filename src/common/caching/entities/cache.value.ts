import { Constants } from "@elrondnetwork/nestjs-microservice-template";

export class CacheValue {
  value?: string;
  ttl: number = Constants.oneSecond() * 6;
}
