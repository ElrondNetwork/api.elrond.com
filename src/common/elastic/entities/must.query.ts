import { AbstractQuery } from "./abstract.query";

export class MustQuery extends AbstractQuery {
  constructor(private readonly queries: AbstractQuery[]) {
    super();
  }

  getQuery(): any {
    return {
      bool: {
        must: this.queries.map(query => query.getQuery()),
      },
    };
  }
}
