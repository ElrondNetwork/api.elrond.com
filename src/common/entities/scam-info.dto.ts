import { ApiProperty } from '@nestjs/swagger';
import { ScamType } from './scam-type.enum';

export class ScamInfo {
  @ApiProperty({ enum: ScamType })
  type: ScamType = ScamType.none;

  @ApiProperty({ type: String, nullable: true })
  info?: string | null;

  static isScam(scamInfo: ScamInfo): boolean {
    return scamInfo.type !== ScamType.none;
  }

  static none(): ScamInfo {
    return {
      type: ScamType.none,
    };
  }
}
