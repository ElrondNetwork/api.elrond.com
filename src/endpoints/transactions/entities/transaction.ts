import { ApiProperty } from "@nestjs/swagger";
import { ScamInfo } from "src/common/entities/scam-info.dto";
import { TransactionType } from "src/endpoints/transactions/entities/transaction.type";
import { TransactionAction } from "../transaction-action/entities/transaction.action";

export class Transaction {
  @ApiProperty({ type: String })
  txHash: string = '';

  @ApiProperty({ type: Number, nullable: true })
  gasLimit: number | undefined = undefined;

  @ApiProperty({ type: Number, nullable: true })
  gasPrice: number | undefined = undefined;

  @ApiProperty({ type: Number, nullable: true })
  gasUsed: number | undefined = undefined;

  @ApiProperty({ type: String, nullable: true })
  miniBlockHash: string | undefined = undefined;

  @ApiProperty({ type: Number, nullable: true })
  nonce: number | undefined = undefined;

  @ApiProperty({ type: String })
  receiver: string = '';

  @ApiProperty({ type: Number })
  receiverShard: number = 0;

  @ApiProperty({ type: Number, nullable: true })
  round: number | undefined = undefined;

  @ApiProperty({ type: String })
  sender: string = '';

  @ApiProperty({ type: Number })
  senderShard: number = 0;

  @ApiProperty({ type: String, nullable: true })
  signature: string | undefined = undefined;

  @ApiProperty({ type: String })
  status: string = '';

  @ApiProperty({ type: String })
  value: string = '';

  @ApiProperty({ type: String, nullable: true })
  fee: string | undefined = undefined;

  @ApiProperty({ type: Number })
  timestamp: number = 0;

  @ApiProperty({ type: String, nullable: true })
  data: string | undefined = undefined;

  @ApiProperty({ type: String, nullable: true })
  function: string | undefined = undefined;

  @ApiProperty({ type: TransactionAction, nullable: true })
  action: TransactionAction | undefined = undefined;

  @ApiProperty({ type: ScamInfo, nullable: true })
  scamInfo: ScamInfo | undefined = undefined;

  @ApiProperty({ enum: TransactionType, nullable: true })
  type: TransactionType | undefined = undefined;

  @ApiProperty({ type: String, nullable: true })
  originalTxHash: string | undefined = undefined;

  @ApiProperty({ type: Boolean, nullable: true })
  pendingResults: boolean | undefined = undefined;

  getDate(): Date | undefined {
    if (this.timestamp) {
      return new Date(this.timestamp * 1000);
    }

    return undefined;
  }
}
