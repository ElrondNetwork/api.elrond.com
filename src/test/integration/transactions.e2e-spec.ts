import { Test } from '@nestjs/testing';
import { PublicAppModule } from 'src/public.app.module';
import { Transaction } from 'src/endpoints/transactions/entities/transaction';
import { TransactionStatus } from 'src/endpoints/transactions/entities/transaction.status';
import { TransactionService } from 'src/endpoints/transactions/transaction.service';
import { TransactionFilter } from 'src/endpoints/transactions/entities/transaction.filter';
import Initializer from './e2e-init';
import { Constants } from 'src/utils/constants';
import { QueryConditionOptions } from 'src/common/elastic/entities/query.condition.options';
import { TransactionOptionalFieldOption } from 'src/endpoints/transactions/entities/transaction.optional.field.options';
import transactionDetails from "../mocks/transactions/transactionDetails";

describe('Transaction Service', () => {
  let transactionService: TransactionService;
  let transactionHash: string;
  let transactionSender: string;
  let transactionReceiver: string;
  const detailedTransactionHash: string = '18128acfd3f19f7a747ccf02bc866e95aa2db92af44fed2f9ed2c2102223b462';

  beforeAll(async () => {
    await Initializer.initialize();

    const moduleRef = await Test.createTestingModule({
      imports: [PublicAppModule],
    }).compile();

    transactionService = moduleRef.get<TransactionService>(TransactionService);

    const transactionFilter = new TransactionFilter();

    const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 1 });
    expect(transactions).toHaveLength(1);

    const transaction = transactions[0];
    transactionHash = transaction.txHash;
    transactionSender = transaction.sender;
    transactionReceiver = transaction.receiver;
  }, Constants.oneHour() * 1000);

  describe('Transactions list', () => {
    it('transactions should have txHash, sender and receiver', async () => {
      const transactionFilter = new TransactionFilter();
      const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });

      for (const transaction of transactions) {
        expect(transaction).toHaveProperty('txHash');
        expect(transaction).toHaveProperty('sender');
        expect(transaction).toHaveProperty('receiver');
      }
    });

    describe('Transactions pagination', () => {
      it(`should return a list with 25 transactions`, async () => {
        const transactionFilter = new TransactionFilter();
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });

        expect(transactions).toBeInstanceOf(Array);
        expect(transactions).toHaveLength(25);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
        }
      });

      it(`should return a list with 100 transactions`, async () => {
        const transactionFilter = new TransactionFilter();
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 100 });

        expect(transactions).toBeInstanceOf(Array);
        expect(transactions).toHaveLength(100);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
        }
      });
    });

    describe('Transactions filters', () => {
      it(`should return a list of transactions between two accounts`, async () => {
        const transactionFilter = new TransactionFilter();
        transactionFilter.sender = transactionSender;
        transactionFilter.receiver = transactionReceiver;
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });

        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          expect(transaction.sender).toBe(transactionSender);
          expect(transaction.receiver).toBe(transactionReceiver);
        }
      });

      it(`should return a list with pending transactions`, async () => {
        const transactionFilter = new TransactionFilter();
        transactionFilter.status = TransactionStatus.pending;
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          expect(transaction.status).toBe(TransactionStatus.pending);
        }
      });

      it(`should return a list with transactions in one date range`, async () => {
        const transactionFilter = new TransactionFilter();
        transactionFilter.before = 1625559162;
        transactionFilter.after = 1625559108;
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          expect(transaction.timestamp).toBeGreaterThanOrEqual(transactionFilter.after);
          expect(transaction.timestamp).toBeLessThanOrEqual(transactionFilter.before);
        }
      });

      it(`should return a list with transactions after one date`, async () => {
        const transactionFilter = new TransactionFilter();
        transactionFilter.after = 1625559108;
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          expect(transaction.timestamp).toBeGreaterThanOrEqual(transactionFilter.after);
        }
      });

      it(`should return a list with transactions before one date`, async () => {
        const transactionFilter = new TransactionFilter();
        transactionFilter.before = 1625559108;
        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          expect(transaction.timestamp).toBeLessThanOrEqual(transactionFilter.before);
        }
      });

      it(`should return transactions for an address`, async () => {
        const address = transactionSender;
        const transactionFilter = new TransactionFilter();
        transactionFilter.sender = address;
        transactionFilter.receiver = address;
        transactionFilter.condition = QueryConditionOptions.should;

        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          if (transaction.sender !== address && transaction.receiver !== address) {
            expect(false);
          }
        }

        const accountTransactionsList = await transactionService.getTransactions(new TransactionFilter(), { from: 0, size: 25 }, undefined, address);
        expect(transactions).toEqual(accountTransactionsList);
      });

      it(`should return transactions for an address with self transactions`, async () => {
        const address = transactionSender;
        const transactionFilter = new TransactionFilter();
        transactionFilter.sender = address;
        transactionFilter.receiver = address;

        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 }, undefined, address);
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          /*delete transaction.action;
          delete transaction.scamInfo;*/

          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          if (transaction.sender !== address || transaction.receiver !== address) {
            expect(false);
          }
        }
      });

      it(`should return a list with transactions where an address is sender, in one date range, with success status`, async () => {
        const address = transactionSender;
        const transactionFilter = new TransactionFilter();
        transactionFilter.after = 1625559108;
        transactionFilter.sender = address;
        transactionFilter.status = TransactionStatus.success;

        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toBeInstanceOf(Array);

        for (const transaction of transactions) {
          expect(transaction).toHaveStructure(Object.keys(new Transaction()));
          expect(transaction.sender).toBe(address);
          expect(transaction.timestamp).toBeGreaterThanOrEqual(transactionFilter.after);
          expect(transaction.status).toBe(TransactionStatus.success);
        }
      });

      it(`should return transactions with specific hashes`, async () => {
        const hashes = '8149581fe858edf8971a73491ff4b26ce2532aa7951ffefafb7b7823ffacc182,56bdbc1a2e9e4dd60bb77c82a72c5b2b77ef51b8decf97f4024fa223b9b64777,INVALIDTXHASH';
        const transactionFilter = new TransactionFilter();
        transactionFilter.hashes = hashes.split(',');

        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 25 });
        expect(transactions).toHaveLength(2);
        const transactionsHashes = transactions.map(({ txHash }) => txHash);
        expect(hashes.split(',').toString()).not.toStrictEqual(transactionsHashes.toString());
      });

      it(`should return transaction details`, async () => {
        const transactionFilter = new TransactionFilter();
        transactionFilter.token = transactionDetails.tokenIdentifier;
        transactionFilter.receiver = transactionDetails.receiver;

        const transactions = await transactionService.getTransactions(transactionFilter, { from: 0, size: 1 });
        for (const transaction of transactions) {
          expect(transaction).toBeDefined();
          expect(transaction.txHash).toMatchSnapshot(transactionDetails.txHash);
          expect(transaction.sender).toMatchSnapshot(transactionDetails.sender);
        }
      });
    });

  });

  describe('Transaction count', () => {
    it(`should return transaction count based on token indentifier`, async () => {
      const transactionFilter = new TransactionFilter();
      transactionFilter.token = transactionDetails.tokenIdentifier;

      const count = await transactionService.getTransactionCount(transactionFilter);
      expect(typeof count).toBe('number');
    });
  });

  describe('Specific transaction', () => {
    it(`should return a transaction for a specific hash`, async () => {
      const transaction = await transactionService.getTransaction(transactionDetails.txHash);
      expect(transaction).toBeInstanceOf(Object);
    });
    it(`should return a transaction for a specific hash with results optional field`, async () => {
      const transaction = await transactionService.getTransaction(transactionDetails.txHash, [TransactionOptionalFieldOption.results]);

      if (transaction) {
        expect(transaction.txHash).toMatchSnapshot(transactionDetails.txHash);
        expect(transaction.results).toBeDefined();
        expect(transaction.receipt).toBeUndefined();
        expect(transaction.logs).toBeUndefined();
      }
      else {
        expect(false);
      }
    });


    it(`should return a transaction for a specific hash with results and logs optional fields`, async () => {
      const transaction = await transactionService.getTransaction(detailedTransactionHash, [TransactionOptionalFieldOption.results, TransactionOptionalFieldOption.logs]);

      if (transaction) {
        expect(transaction.txHash).toMatchSnapshot(transactionDetails.txHash);
        expect(transaction.results).toBeDefined();
        expect(transaction.receipt).toBeUndefined();
        expect(transaction.operations).toBeDefined();
      }
      else {
        expect(false);
      }
    });

    it(`should throw 'Transaction not found' error`, async () => {
      expect(await transactionService.getTransaction(transactionHash + 'a')).toBeNull();
    });
  });

  describe('Get Transaction Count For Address', () => {
    it('should return transaction count for address', async  () => {
      const address = transactionDetails.sender;
      const count = await transactionService.getTransactionCountForAddress(address);

      return expect(typeof count).toBe('number');
    });
  });

  describe('Get Transaction Count For Address Raw', () => {
    it('should return transaction count for address raw', async  () => {
      const addressRaw = transactionDetails.sender;
      const count = await transactionService.getTransactionCountForAddressRaw(addressRaw);

      return expect(typeof count).toBe('number');
    });
  });
});
