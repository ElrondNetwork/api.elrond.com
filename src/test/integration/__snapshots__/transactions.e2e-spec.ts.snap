// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Service Specific transaction should return a transaction for a specific hash with results and logs optional fields: 87db96d4a04dc5729049a3faf6e7cd6064ecc91f48bdbef3a7889b8fb61538e8 1`] = `"18128acfd3f19f7a747ccf02bc866e95aa2db92af44fed2f9ed2c2102223b462"`;

exports[`Transaction Service Specific transaction should return a transaction for a specific hash with results optional field: 87db96d4a04dc5729049a3faf6e7cd6064ecc91f48bdbef3a7889b8fb61538e8 1`] = `"87db96d4a04dc5729049a3faf6e7cd6064ecc91f48bdbef3a7889b8fb61538e8"`;

exports[`Transaction Service Transactions list Transactions filters should return transaction details: 87db96d4a04dc5729049a3faf6e7cd6064ecc91f48bdbef3a7889b8fb61538e8 1`] = `"87db96d4a04dc5729049a3faf6e7cd6064ecc91f48bdbef3a7889b8fb61538e8"`;

exports[`Transaction Service Transactions list Transactions filters should return transaction details: erd137esrr64ker9cspja6ey4wna7kqau3fyzawk9xgyr9n6925u6f2sm2rja6 1`] = `"erd137esrr64ker9cspja6ey4wna7kqau3fyzawk9xgyr9n6925u6f2sm2rja6"`;
